source/common/unified_memory_allocation/include/uma/memory_provider.h

```diff
@@ -75,6 +75,16 @@ enum uma_result_t umaMemoryProviderFree(uma_memory_provider_handle_t hProvider,
 ///         adapter specific result is an error.
 enum uma_result_t umaMemoryProviderGetLastResult(uma_memory_provider_handle_t hProvider, const char **ppMessage);
 
+///
+/// \brief Create a proxy memory provider that associates each allocation with a pool
+///        and forwards calls to the upstream provider. By calling umaPoolByPtr()
+///        it's possible to obtain pool associated with the pointer.
+/// \param hUpstream all calls are forwarded to this provider
+/// \param pool pool to be associated with all allocations
+/// \param [out] returns instance of tracking provider
+enum uma_result_t umaTrackingMemoryProviderCreate(uma_memory_provider_handle_t hUpstream,
+                                                  void *pool, uma_memory_provider_handle_t *hTrackingProvider);
```

Comment:
Why `void*` instead of `uma_memory_pool_handle_t`?

Reply:
I've changed it to `uma_memory_pool_handle_t` but I'm not sure about this one. In current design this pointer is what user gets within `pool::initialize` function as `void** pool`. If we make it `uma_memory_pool_handle_t` the user needs to do casting.